src/server/system_services/bucket_server.js

```diff
@@ -1900,6 +1948,79 @@ function validate_non_nsfs_bucket_creation(req) {
     }
 }
 
+// updates replication policy of all bucket class buckets
+async function update_bucket_class_replication(req) {
```

Comment:
I think you can refactor such that both `update_bucket_class_replication` and `put_bucket_replication` can use the same internal function to update a replication policy

Reply:
merged the logic, thanks